File: Android/app/src/main/kotlin/com/zhechengqi/tollcat/settings/TipHistoryItem.kt
package com.zhechengqi.tollcat.settings

data class TipHistoryItem(
    val transactionID: String,
    val productID: String,
    val displayPrice: String,
    val purchasedAtMillis: Long,
    val name: String?,
    val message: String?,
    val isSubmitted: Boolean,
    val jws: String,
    val appVersion: String,
) {
    val productTitleRes: Int
        get() = TipProductID.fromRaw(productID)?.titleRes ?: 0
}

File: Packages/MeterKit/Sources/MeterFeatures/Settings/TipHistoryItem.swift
import Foundation
import MeterTips

struct TipHistoryItem: Identifiable, Hashable, Sendable {
    var transactionID: String
    var productID: String
    var displayPrice: String
    var purchasedAt: Date
    var name: String?
    var message: String?
    var isSubmitted: Bool

    var id: String { transactionID }

    var productTitle: String {
        TipProductID(rawValue: productID)?.listTitle ?? productID
    }
}
